canvas_modules/common-canvas/src/common-properties/controls/checkboxset/checkboxset.jsx

```diff
@@ -74,27 +78,52 @@ class CheckboxsetControl extends React.Component {
 	}
 
 	render() {
+
+		const hidden = this.props.state === STATES.HIDDEN;
+		const disabledProps = this.props.state === STATES.DISABLED;
+
 		let controlValue = this.props.value;
 		if (typeof controlValue === "undefined" || controlValue === null) {
 			controlValue = [];
 		}
 		const checkboxes = [];
 		for (var i = 0; i < this.props.control.values.length; i++) {
+			if (this.props.control.valueDescs && !this.props.tableControl) {
```

Comment:
It would be better to wrap the entire tooltipIcon in an if check like this

let tooltipIcon = null;
if (Array.isArray(this.props.control.valueDescs) && !isEmpty(this.props.control.valueDescs[i])) {
...
}

Reply:
Updated the logic.

Reply:
This isn't correct yet.  Why set the tooltip here instead of in the if check where `tooltipIcon` is set?

Reply:
Why not move this into this if check `if (Array.isArray(this.props.control.valueDescs) && !isEmpty(this.props.control.valueDescs[i])) {`.  Then you only need 1 if check.

Reply:
Left only one if check.